static/docs/commands-reference/fetch.md

```diff
@@ -1,62 +1,243 @@
+Note, `dvc push`, `dvc fetch`, and `dvc pull` are related in that these 3
+commands perform data synchronization among local and remote storage. The
+specific way in which the set of files to push/fetch/pull is determined
+begins with calculating the checksums of the files in question, when these
+are [added](https://dvc.org/doc/get-started/add-files) to DVC. File checksums
+are then stored in the corresponding DVC files (usually saved in a Git
+branch). Only the checksums specified in DVC files currently in the workspace
+are considered by `dvc fetch` (unless the `-a` or `-T` options are used).
+
+The default remote is used unless `--remote` is specified. See `dvc remote add`
+for more information on how to configure different remote storage providers.
+
+Fetching brings files in from a remote to the local cache, making them 
+available for linking (or copying) into the workspace. (Refer to
+[dvc config cache.type](https://dvc.org/doc/commands-reference/config#cache).)
+
+Note, `dvc fetch` is performed automatically by `dvc pull` when the target stage
+files are not already in the local cache. Specifically,
+`dvc pull` = `dvc fetch` + `dev checkout`:
 
 ```
+Controlled files             Commands
+---------------- ---------------------------------
+
+ remote storage
+       +
+       |         +------------+ 
+       | - - - - | dvc fetch  | ++
+       v         +------------+   +   +----------+
+  local cache                      ++ | dvc pull |
+       +         +------------+   +   +----------+
+       | - - - - |dvc checkout| ++
+       |         +------------+
+       v
+   workspace
+```
+
+## Options
+
+- `-r REMOTE`, `--remote REMOTE` - name of the
+  [remote storage](https://dvc.org/doc/commands-reference/remote#description)
+  to fetch from (see `dvc remote list`). If not specified, the default 
+  remote is used (see `dvc config core.remote`). The argument `REMOTE` is a
+  remote name defined using the `dvc remote` command.
+
+- `-d`, `--with-deps` - fetch cache by tracking dependencies to the named
+  target stages. This option only has effect when one or more `targets` are
+  specified. By traversing the dependencies, DVC searches backward through the
+  pipeline from the named target(s). This means DVC will not fetch files needed
+  later in the pipeline than the named target(s).
+
+- `-R`, `--recursive` - this option tells DVC that `targets` are
+  directories (not DVC files), and to traverse them recursively. All DVC 
+  files found will be read in order to determine the set of data files to fetch.
+
+- `-j JOBS`, `--jobs JOBS` - number of threads to run simultaneously to 
+  handle the downloading of files from the remote. Using more jobs may 
+  improve the total download speed if a combination of small and large files 
+  are being fetched.
+  The default value is `4 * cpu_count()`. For SSH remotes default is 4.
+  > Note, this applies only to local remotes and for SSH remotes. (See `dvc 
```

Comment:
I'm not sure I understand this `Note`

Reply:
Updated to
> Note, this option applies only to fetching from local remotes or from SSH  remotes. (See `dvc remote` for more information about remotes.)

Reply:
I'm just not sure that this statement is correct. Would you mind to confirm it on Discord?

Reply:
Yes, I confirmed this some time ago (: 
![image](https://user-images.githubusercontent.com/1477535/55246634-f3279c00-5288-11e9-9957-92f4cabbf4a7.png)

Reply:
@mroutis @jorgeorpinel --jobs applies to all remotes(including s3) though. For fetch, it determines how many thread workers are going to be used to download data from remote.

Reply:
Thanks @efiop 

What are the default jobs for all the other types of remotes? Like S3

Reply:
@jorgeorpinel it is 4*cpu_count(), so basically 4 times the number of cores that a machine has.

Reply:
OK, note removed then (: